<!DOCTYPE html>
<html lang="en">
<head>
  
  
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    
    
    <link rel="canonical" href="https://opensciencegrid.org/virtual-school-pilot-2020/materials/sw/part1-ex1-download/">
    <link rel="shortcut icon" href="../../../img/favicon.ico">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
    <title>Exercise 1.1 - OSG Virtual School Pilot 2020</title>
    <link href="../../../css/bootstrap-3.3.7.min.css" rel="stylesheet">
    <link href="../../../css/font-awesome-4.7.0.css" rel="stylesheet">
    <link href="../../../css/base.css" rel="stylesheet">
    <link rel="stylesheet" href="../../../css/highlight.css">
    <link href="../../../css/extra.css" rel="stylesheet">
    <link href="../../../css/codehilite.css" rel="stylesheet">
    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
        <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
        <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <script src="../../../js/jquery-3.2.1.min.js"></script>
    <script src="../../../js/bootstrap-3.3.7.min.js"></script>
    <script src="../../../js/highlight.pack.js"></script>
    
    <base target="_top">
    <script>
      var base_url = '../../..';
      var is_top_frame = false;
        
        var pageToc = [
          {title: "Software Exercise 1.1: Using a Pre-compiled Binary", url: "#_top", children: [
              {title: "Background", url: "#background" },
              {title: "Our Software Example", url: "#our-software-example" },
              {title: "Copy the Input Files", url: "#copy-the-input-files" },
              {title: "Submitting the Job", url: "#submitting-the-job" },
          ]},
        ];

    </script>
    <script src="../../../js/base.js"></script> 
</head>

<body>
<script>
if (is_top_frame) { $('body').addClass('wm-top-page'); }
</script>



<div class="container-fluid wm-page-content">
  <a name="_top"></a>
    

    <style type="text/css"> pre em { font-style: normal; background-color: yellow; } pre strong { font-style: normal; font-weight: bold; color: \#008; } </style>

<h1 id="software-exercise-11-using-a-pre-compiled-binary">Software Exercise 1.1: Using a Pre-compiled Binary<a class="headerlink" href="#software-exercise-11-using-a-pre-compiled-binary" title="Permanent link">&para;</a></h1>
<p>In this exercise, we will run a job using a downloaded, pre-compiled binary. This exercise should take 10-15 minutes.</p>
<h2 id="background">Background<a class="headerlink" href="#background" title="Permanent link">&para;</a></h2>
<p>This is the simplest scenario for using a particular software program on the Open Science Grid - downloading a pre-compiled binary and using it to run jobs. </p>
<h2 id="our-software-example">Our Software Example<a class="headerlink" href="#our-software-example" title="Permanent link">&para;</a></h2>
<p>The software we will be using for this example is a common tool for aligning genome and protein sequences against a
reference database, the BLAST program.</p>
<ol>
<li>
<p>Search the internet for the BLAST software.  Searches might include "blast executable or "download blast software".  Hopefully these searches will lead you to a BLAST website page that looks like this:</p>
<p><img alt="BLAST landing page" src="../../../materials/sw/files/part1-ex1-blast-landing-page.png" /></p>
</li>
<li>
<p>Click on the title that says <a href="../../../materials/sw/files/part1-ex1-blast-front-page.png">"Download BLAST"</a> and then look for the link that has the <a href="../../../materials/sw/files/part1-ex1-blast-dl-page.png">latest installation and source code</a>.  You should end up on a page with a list of each version of BLAST that is available for different operating systems.</p>
</li>
<li>
<p>We could download the source and compile it ourselves, but instead, we're going to use one of the pre-built binaries.  Before proceeding, look at the list of downloads and try to determine which one you want. </p>
</li>
<li>
<p>Based on our operating system, we want to use the Linux binary, which is labelled with the <code>x64-linux</code> suffix. </p>
<p><img alt="BLAST downloads" src="../../../materials/sw/files/part1-ex1-blast-src-page.png" /></p>
<p>All the other links are either for source code or other operating systems. </p>
</li>
<li>
<p>While logged into <code>login05.osgconnect.net</code>, create a directory for this exercise. Then download the appropriate <code>tar.gz</code> file and un-tar it. You can download the file directly from the BLAST website using <code>wget</code> or download our local copy with the command below: </p>
<div class="codehilite"><pre><span></span><span class="gp">user@login $</span> wget http://proxy.chtc.wisc.edu/SQUID/osgschool19/ncbi-blast-2.10.1+-x64-linux.tar.gz
<span class="gp">user@login $</span> tar -xzf ncbi-blast-2.10.1+-x64-linux.tar.gz
</pre></div>


</li>
<li>
<p>We're going to be using the <code>blastx</code> binary in our job. Where is it in the directory you just downloaded?</p>
</li>
</ol>
<h2 id="copy-the-input-files">Copy the Input Files<a class="headerlink" href="#copy-the-input-files" title="Permanent link">&para;</a></h2>
<p>To run BLAST, we need an input file and reference database. For this example, we'll use the "pdbaa" database, which contains sequences for the protein structure from the Protein Data Bank. For our input file, we'll use an abbreviated fasta file with mouse genome information.</p>
<ol>
<li>
<p>Download these files to your current directory: </p>
<div class="codehilite"><pre><span></span><span class="gp">username@login $</span> wget http://proxy.chtc.wisc.edu/SQUID/osgschool20/pdbaa.tar.gz
<span class="gp">username@login $</span> wget http://proxy.chtc.wisc.edu/SQUID/osgschool20/mouse.fa
</pre></div>


</li>
<li>
<p>Untar the <code>pdbaa</code> database: </p>
<div class="codehilite"><pre><span></span><span class="gp">username@login $</span> tar -xzf pdbaa.tar.gz
</pre></div>


</li>
</ol>
<h2 id="submitting-the-job">Submitting the Job<a class="headerlink" href="#submitting-the-job" title="Permanent link">&para;</a></h2>
<p>We now have our program (the pre-compiled <code>blastx</code> binary) and our input files, so all that remains is to create the submit file. The form of a typical <code>blastx</code> command looks something like this:</p>
<div class="codehilite"><pre><span></span>blastx -db &lt;database_dir/prefix&gt; -query &lt;input_file&gt; -out &lt;output_file&gt;
</pre></div>


<ol>
<li>
<p>Copy a submit file from one of the intro exercises to use for this exercise. </p>
</li>
<li>
<p>Think about which lines you will need to change or add to your submit file in order to submit the job successfully. In particular:    </p>
<ul>
<li>What is the executable?</li>
<li>How can you indicate the entire command line sequence above?</li>
<li>Which files need to be transferred in addition to the executable?</li>
<li>Does this job require a certain type of operating system?</li>
<li>Do you have any idea how much memory or disk to request?</li>
</ul>
</li>
<li>
<p>Try to answer these questions and modify your submit file appropriately.</p>
</li>
<li>
<p>Once you have done all you can, check your submit file against the lines below, which contain the exact components to run this particular job.</p>
<ul>
<li>
<p>The executable is <code>blastx</code>, which is located in the <code>bin</code> directory of our downloaded BLAST directory. We need to use the <code>arguments</code> line in the submit file to express the rest of the command. </p>
<div class="codehilite"><pre><span></span>executable = ncbi-blast-2.10.1+/bin/blastx
arguments = -db pdbaa/pdbaa -query mouse.fa -out results.txt
</pre></div>


</li>
<li>
<p>The BLAST program requires our input file and database, so they must be transferred with <code>transfer_input_files</code>. </p>
<div class="codehilite"><pre><span></span>transfer_input_files = pdbaa, mouse.fa
</pre></div>


</li>
<li>
<p>Let's assume that we've run this program before, and we know that 1GB of disk and 1GB of memory will be MORE than enough (the 'log' file will tell us how accurate we are, after the job runs): </p>
<div class="codehilite"><pre><span></span>request_memory = 1GB
request_disk = 1GB
</pre></div>


</li>
<li>
<p>Because we downloaded a Linux-specific binary, we need to request machines that are running Linux. </p>
<div class="codehilite"><pre><span></span>requirements = (OSGVO_OS_STRING == &quot;RHEL 7&quot;)
</pre></div>


</li>
</ul>
</li>
<li>
<p>Submit the blast job using <code>condor_submit</code>. Once the job starts, it should run in just a few minutes and produce a file called <code>results.txt</code>.</p>
</li>
</ol>

  <br>
    

  <div class="row wm-article-nav-buttons" role="navigation" aria-label="navigation">
    
    <div class="wm-article-nav pull-right">
      <a href="../part1-ex2-wrapper/" class="btn btn-xs btn-default pull-right">
        Next
        <i class="fa fa-chevron-right" aria-hidden="true"></i>
      </a>
      <a href="../part1-ex2-wrapper/" class="btn btn-xs btn-link">
        Exercise 1.2
      </a>
    </div>
    
    <div class="wm-article-nav">
      <a href="../../osg/ex5-software-diffs/" class="btn btn-xs btn-default pull-left">
        <i class="fa fa-chevron-left" aria-hidden="true"></i>
        Previous</a><a href="../../osg/ex5-software-diffs/" class="btn btn-xs btn-link">
        Exercise 5
      </a>
    </div>
    
  </div>

    <br>
</div>

<footer class="col-md-12 wm-page-content">
  <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a> using <a href="https://github.com/gristlabs/mkdocs-windmill">Windmill</a> theme by Grist Labs.</p>
</footer>

</body>
</html>